tests/benchmarks/sched_queues/src/main.c

```diff
@@ -0,0 +1,308 @@
+/*
+ * Copyright (c) 2024 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/*
+ * @file
+ * This file contains the main testing module that invokes all the tests.
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/timestamp.h>
+#include "utils.h"
+#include <zephyr/tc_util.h>
+#include <ksched.h>
+
+#define TEST_STACK_SIZE (1024 + CONFIG_TEST_EXTRA_STACK_SIZE)
+#define BUSY_STACK_SIZE (1024 + CONFIG_TEST_EXTRA_STACK_SIZE)
+
+uint32_t tm_off;
+
+/*
+ * Warning! Most of the created threads in this test use the same stack!
+ * This is done to reduce the memory footprint as having unique stacks
+ * for hundreds or thousands of threads would require substantial memory.
+ * We can get away with this approach as the threads sharing the same
+ * stack will not be executing, even though they will be ready to run.
+ */
+
+static K_THREAD_STACK_DEFINE(test_stack, TEST_STACK_SIZE);
+
+#if CONFIG_MP_MAX_NUM_CPUS > 1
+K_THREAD_STACK_ARRAY_DEFINE(busy_stack, CONFIG_MP_MAX_NUM_CPUS - 1, BUSY_STACK_SIZE);
+static struct k_thread busy_thread[CONFIG_MP_MAX_NUM_CPUS - 1];
+#endif
+
+static struct k_thread test_thread[CONFIG_BENCHMARK_NUM_THREADS];
+
+static uint64_t add_cycles[CONFIG_BENCHMARK_NUM_THREADS];
+static uint64_t remove_cycles[CONFIG_BENCHMARK_NUM_THREADS];
+
+extern void z_unready_thread(struct k_thread *thread);
+
+#if CONFIG_MP_MAX_NUM_CPUS > 1
+static void busy_entry(void *p1, void *p2, void *p3)
+{
+	ARG_UNUSED(p1);
+	ARG_UNUSED(p2);
+	ARG_UNUSED(p3);
+
+	while (1) {
+	}
+}
+#endif
+
+/**
+ * The test entry routine is not expected to execute.
+ */
+static void test_entry(void *p1, void *p2, void *p3)
+{
+	ARG_UNUSED(p2);
+	ARG_UNUSED(p3);
+
+	printk("Thread %u unexpectedly executed\n",
+	       (unsigned int)(uintptr_t)p1);
+
+	while (1) {
+	}
+}
+
+static void start_threads(unsigned int num_threads)
+{
+	unsigned int i;
+	unsigned int bucket_size;
+
+#if CONFIG_MP_MAX_NUM_CPUS > 1
```

Comment:
Similar comment: no need for preprocessor test, if MP_MAX_NUM_CPUS == 0 then the compiler will remove that loop for you.